### DISCUSSION TITLE
Get full state dynamically

### DISCUSSION THREAD

**Discussion by Mardank** [USC-ACTLab/crazyswarm#663] [Q&A] [answered]
Hey all,

recently I started to work into logging with crazyswarm and ROS. So far I come up with the following approaches 

1. Defining a ROS Topic in the launch file to log the parameters via console.
2. Logging only the position using the function position().

Now my question is, whether it is possible to get more parameters  like the full State of the drone during the flight? Is there a way to get these logging parameters in python while flying?

I thought about writing a method similar to position for the full state, but this would require some heavy modification of the 'crazyswarm_server.cpp' file.

Kind regards,
Markus K

**Comment by whoenig** [comment, marked as answer]
If you just want the full pose, you can also [enable logging of the pose](https://github.com/USC-ACTLab/crazyswarm/blob/master/ros_ws/src/crazyswarm/launch/hover_swarm.launch#L44-L45). Then, there will be a ROS topic of the form `/cf<x>/pose`. If you want custom logging variables, you use the method 1, that you described.

In order to use the result in your Python code, you have to subscribe to the ROS topic on the Python side - no modification of the C++ code needed. A subscriber will get a callback whenever new data is available (see the ROS documentation/tutorial).

**Reply by Mardank** [reply]
Thanks, for the answer.
